src/Features/Core/Portable/RemoveUnusedVariable/AbstractRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -39,6 +39,14 @@ internal abstract class AbstractRemoveUnusedVariableCodeFixProvider<TLocalDeclar
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var diagnostic = context.Diagnostics.First();
+
+            // bail out if there are syntax errors, other than the ones that we're trying to fix
+            var treeDiagnostics = diagnostic.Location.SourceTree.GetDiagnostics(context.CancellationToken);
+            if (treeDiagnostics.Any(d => d.Severity == DiagnosticSeverity.Error && !FixableDiagnosticIds.Contains(d.Id)))
```

Comment:
I would have assumed the fixable diagnostic IDs in this case were always semantic diagnostics, which you wouldn't get the from the tree?

Reply:
yeah, i'm not a fan of this change.  where is the actual crash happening, and why?  checking for syntax-errors is just too broad a brush.

Reply:
> I would have assumed the fixable diagnostic IDs in this case were always semantic diagnostics, which you wouldn't get the from the tree?

Thank you, I wasn't aware there were two types, though now that you've said it it makes perfect sense. As always, I just found some code somewhere that seemed to do the job and copied it 😄

> checking for syntax-errors is just too broad a brush.

I don't disagree, but the targeted fix felt _too_ targeted, to approaching ridiculous.

> where is the actual crash happening, and why?

So `if (test and test) { }` is parsed such that `and test` is the unused variable wanting to be removed, and is the statement of the if. The braces after a separate and unrelated `BlockSyntax`. So when removing `and test` the syntax node is (rightly) throwing because its being asked to construct an if with no statement.

Targeting the fix to "ignore if the local variable is the only statement in an if" would certainly work, but also feels too specific as it does nothing for `while (test and test) { }` or `switch (test and test) { }` etc.

Reply:
so a narrow check for me would be for us to simply check 'is this statement parented by a statement list?'  if not, we shouldn't remove it. 

basically, teh parent must be an 'executable block' (which is either a BlockStatement or a Case-Clause-Thingy).  I believe we have a helper for this.